### PR TITLE
Release/wanderer

### PR DESCRIPTION
**Pull request by jtbaird**
# PR Details
## Description
This PR is to merge the Wanderer release (carma-system-3.4.0) into the master branch.

## Related Issue
None

## Motivation and Context
Purpose of the Wanderer release is to incorporate four major enhancements into the CARMA system. First, a steering limiter was added in order to prevent Carma from taking turns that present a risk of vehicle rollover. Second, updates were made to the health_monitor to allow for graceful handling of sensor failure, including control handover to the human driver. Third, carma-messenger was updated to detect nearby automated vehicles via DSRC. And fourth, functionality was added to carma-messenger to allow for requesting, receiving, and displaying truck safety info from a passing Carma truck via DSRC while vehicles are in motion. Along with the above enhancements, several bug fixes and CI related enhancements are included in this release.

## How Has This Been Tested?
Per the release test plan.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,16 +49,18 @@ jobs:
...
           name: Build Driver
           command: |
             source ${INIT_ENV}
+            export ROS_PARALLEL_JOBS='-j1 -l1' # Try to reduce memory consumption on build
             build-wrapper-linux-x86-64 --out-dir /opt/carma/bw-output bash make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
       - run:
           name: Run C++ Tests
           command: |
             source ${INIT_ENV}
+            export ROS_PARALLEL_JOBS='-j1 -l1' # Try to reduce memory consumption on build
```
Comment (maefromm) [inline .circleci/config.yml:63]: Under line 81, i see " cd src/CARMANovatelGpsDriver"

Comment (jtbaird) [reply .circleci/config.yml:63]: yeah that's still the path

File: .circleci/config.yml [hunk 2]
```diff
@@ -49,16 +49,18 @@ jobs:
           name: Pull Deps
           command: |
             source ${INIT_ENV}
-            ./src/CARMANovatelGpsDriver/docker/checkout.sh ${PWD}
+            ./src/CARMANovatelGpsDriver/docker/checkout.bash -r ${PWD}
```
Comment (maefromm) [inline .circleci/config.yml:52]: Is this path correct?

**Review by maefromm — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (jtbaird) [reply .circleci/config.yml:52]: path is correct

Comment (jtbaird) [reply .circleci/config.yml:52]: that dir that it's running the script in gets created a little higher up in the file

**Review by maefromm — APPROVED** [review]

**Pull request merged**